tasks/generate_token.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+
+# Locations to look for the autosign executable
+executables=(
+    '/opt/puppetlabs/puppetserver/bin/autosign'
+    '/opt/puppetlabs/puppet/bin/autosign'
+    '/opt/puppet/bin/autosign'
+    '/usr/local/bin/autosign'
+)
+
+# Look for the executable and pick the first one we find
```

Comment:
I think this will actually pick the last one we find, since it doesn't break out of the loop. Doesn't matter, but stuck out to me. I think we could skip having to do this at all by prepending all of those paths to `PATH` and then letting bash find an autosign executable.

```bash
PATH="/opt/puppetlabs/puppetserver/bin:/opt/puppetlabs/puppet/bin:/opt/puppet/bin:/usr/local/bin:$PATH"
```

I don't feel super strongly, just a minor optimization.